### PR TITLE
HDDS-6119. Reset the property name to configure the datanode kerberos keytab

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

The current settings to configure the keytab file to be used by the datanode follows the same naming convention as the other components.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6119

## How was this patch tested?

Local hugo test.

### CODE REVIEW

File: hadoop-hdds/docs/content/security/SecuringDatanodes.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ that is setup in  hdfs-site.xml.
 Property|Description
 --------|--------------
 dfs.datanode.kerberos.principal|The datanode service principal. <br/> e.g. dn/_HOST@REALM.COM
-dfs.datanode.keytab.file| The keytab file used by datanode daemon to login as its service principal.
+dfs.datanode.kerberos.keytab.file| The keytab file used by datanode daemon to login as its service principal.
```
Comment (adoroszlai) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.md:41]: Can you please point out where `dfs.datanode.kerberos.keytab.file` is used?

`dfs.datanode.keytab.file` is defined in https://github.com/apache/ozone/blob/d6cbb15f7a29221fcbd8f2c3bc9f5073c12c12d7/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java#L72-L73 and used in https://github.com/apache/ozone/blob/d6cbb15f7a29221fcbd8f2c3bc9f5073c12c12d7/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java#L244-L254

Comment (sky76093016) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.md:41]: Thanks to @adoroszlai ’s review, I think the original intention of this [jira](https://issues.apache.org/jira/projects/HDDS/issues/HDDS-6119) is to change `dfs.datanode.keytab.file` to `dfs.datanode.kerberos.keytab.file`, which is similar to naming conventions such as `dfs.datanode.kerberos.principal`. 
The purpose of this issue is to change its naming for consistency. Do you think this is feasible? If change the naming, the code you mentioned also needs to be changed.

Comment (adoroszlai) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.md:41]: Thanks for the explanation.

I guess we can change the property name, but need to keep backwards compatibility by adding deprecation.

Doc change in itself is not OK, as it introduces mismatch.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java [hunk 2]
```diff
@@ -69,9 +69,13 @@ private DFSConfigKeysLegacy() {
   public static final String DFS_DATANODE_KERBEROS_PRINCIPAL_KEY =
       "dfs.datanode.kerberos.principal";
 
+  @Deprecated
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java:72]: We need to add the pair of old/new keys here:

https://github.com/apache/ozone/blob/d6cbb15f7a29221fcbd8f2c3bc9f5073c12c12d7/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java#L297-L306

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sky76093016 for updating the patch.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java (hunk 2 above)
Comment (sky76093016) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/DFSConfigKeysLegacy.java:72]: Thanks for the reminder, let me add it.

**Comment by sky76093016** [comment]
@adoroszlai Please take a look. 

BTW, do we need to change the title of the PR appropriately for the modified content?

**Review by adoroszlai — APPROVED** [review]

**Review by cku328 — APPROVED** [review]
Thanks @sky76093016 for the patch.

**Pull request merged**